### PR TITLE
Introduce executeQuery on Driver level

### PR DESCRIPTION
**Pull request by injectives**
This is a new basic high-level API for executing idempotent queries.

### CODE REVIEW

File: driver/src/main/java/org/neo4j/driver/Config.java [hunk 1]
```diff
@@ -275,6 +293,19 @@ public static final class ConfigBuilder {
...
         private ConfigBuilder() {}
 
+        /**
+         * Sets a {@link BookmarkManager} implementation for the driver to use on
+         * {@link Driver#executeQuery(Query, QueryConfig)} method and its variants by default.
+         *
+         * @param queryBookmarkManager bookmark manager, must not be {@code null}
+         * @return this builder
+         */
+        public ConfigBuilder withQueryBookmarkManager(BookmarkManager queryBookmarkManager) {
```
Comment (gjmwoods) [inline driver/src/main/java/org/neo4j/driver/Config.java:309]: thinking about the chances users will think configuring this BookmarkManager will apply for the entire driver. Not sure how/if we can reduce that risk.

File: driver/src/main/java/org/neo4j/driver/Driver.java [hunk 2]
```diff
@@ -63,6 +66,109 @@
  * @since 1.0 (Modified and Added {@link AsyncSession} and {@link RxSession} since 2.0)
  */
 public interface Driver extends AutoCloseable {
+    /**
+     * Execute an idempotent query in a managed transaction with automatic retries on retryable errors.
+     * <p>
+     * This is a basic high-level API for executing idempotent queries. {@link #executeQuery(Query, QueryConfig)}
+     * documentation provides more details.
+     *
```
Comment (gjmwoods) [inline driver/src/main/java/org/neo4j/driver/Driver.java:74]: perhaps a note back to bookmark configuration and a note about causal consistency guarantees similar to the one on `Session`

**Review by gjmwoods — APPROVED** [review]

File: driver/src/main/java/org/neo4j/driver/Config.java (hunk 1 above)
Comment (injectives) [reply driver/src/main/java/org/neo4j/driver/Config.java:309]: A note on sessions has been added.

File: driver/src/main/java/org/neo4j/driver/Driver.java (hunk 2 above)
Comment (injectives) [reply driver/src/main/java/org/neo4j/driver/Driver.java:74]: Updated. Please take a look if it matches you expectations.

**Review by gjmwoods — APPROVED** [review]

**Pull request merged**